### PR TITLE
Fix response handler

### PR DESCRIPTION
**Pull request by mjsalinger**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
   "dependencies": {
     "bluebird": "^3.0.5",
     "express": "^4.13.3",
-    "oauth2-server": "next",
+    "oauth2-server": "kinvey/node-oauth2-server#kinvey",
```
Comment (maxtruxa) [inline package.json]: This should probably stay at `"next"`. 😉

Comment (mjsalinger) [inline package.json]: Fixed..

**Comment by maxtruxa** [comment]
Tests fixed with #33.

**Comment by mjsalinger** [comment]
Rebased to remove some duplicate commits and incorporate latest changes to `master`.  @maxtruxa, one final review?

File: index.js [hunk 2]
```diff
@@ -119,10 +119,18 @@ ExpressOAuthServer.prototype.token = function(options) {
...
-  res.set(response.headers);
-  res.status(response.status).send(response.body);
+
+  if (response.status === 302) {
+    var location = response.headers.location;
+    delete response.headers.location;
+    res.set(response.headers);
+    res.redirect(location);
+  }
+  else {
```
Comment (maxtruxa) [inline index.js]: Style-wise.. Are we going by the 1TBS
```
} else {
```
or Stroustrup?
```
}
else {
```

Comment (mjsalinger) [inline index.js]: The former...

**Comment by maxtruxa** [comment]
@mjsalinger 
Are we always using 302 to redirect? Even if so, it might be a good idea to support the other redirect codes.
- 301
- 302
- 303
- 307
- 308

**Comment by mjsalinger** [comment]
302 is the default used in the oauth2 spec..  Others are allowed, and I think that's a good idea to add support for, but I think that should be a separate PR for a future release.

**Comment by mjsalinger** [comment]
Added issue #37 to track.

**Pull request merged**